integration_test/sql/migration.exs

```diff
@@ -65,6 +65,31 @@ defmodule Ecto.Integration.MigrationTest do
     end
   end
 
+  defmodule AlterForeignKeyMigration do
+    use Ecto.Migration
+
+    def up do
+      create table(:alter_fk_migration_users)
+
+      create table(:alter_fk_migration_posts) do
+        add :alter_fk_migration_user_id, :integer
+      end
+
+      alter table(:alter_fk_migration_posts) do
+        modify :alter_fk_migration_user_id, references(:alter_fk_migration_users, on_delete: :nilify_all)
+      end
+
+      execute "INSERT INTO alter_fk_migration_users (id) VALUES ('1')"
+      execute "INSERT INTO alter_fk_migration_posts (id, alter_fk_migration_user_id) VALUES ('1', '1')"
+      execute "DELETE FROM alter_fk_migration_users"
```

Comment:
I might be missing the point why we would want to do that. I am deleting the record to validate that ON CASCADE clause gets executed...